Dialog defaults (#2888)
modelica/ModelicaSpecification: chapters/annotations.tex

An empty \lstinline!group! (the default) means tool-specific choice of group.
The order of components (and class definitions) within each group and the order of the groups and tabs are according to the declaration order, where inherited elements are added at the place of the extends.

> A variable shall have at most one of \lstinline!showStartAttribute=true!, \lstinline!colorSelector=true!, \lstinline!loadSelector!, \lstinline!saveSelector! or \lstinline!connectorSizing=true!.

Reviewer: Shouldn't we say _component_ rather than _variable_?

Author: Yes, updating.

The \lstinline!showStartAttribute = true! is primarily intended for non-parameter values and avoids introducing a separate parameter for the start-value of the variable.
\end{nonnormative}

> If \lstinline!colorSelector = true!, it suggests that an rgb-value selector can be presented for a vector of three elements and generate values 0..255 (the color selector should be useable both for vectors of \lstinline!Integer! and \lstinline!Real!).

Reviewer: If we switch to suggesting things, I think we can be more direct:
```suggestion
If \lstinline!colorSelector = true!, it suggests use of an rgb-value selector to pick a vector of three elements with values in the range 0..255 (the color selector should be useable both for vectors of \lstinline!Integer! and \lstinline!Real!).
```

Author: I prefer the less direct current variant here. The reason is that from the user perspective it should be about selecting a color (which will generate an rgb-vector); not about picking a vector. Clearly it can be reformulated in other ways.

> The presence of \lstinline!loadSelector! or \lstinline!saveSelector! specifying \fmtannotationindex{Selector} suggests a file dialog to select a file.

Reviewer: Looks like some words are missing (at least for the kind of suggestion that I find consistent with the rgb-value selector phrase above):
```suggestion
The presence of \lstinline!loadSelector! or \lstinline!saveSelector! specifying \fmtannotationindex{Selector} suggests use of a file dialog to select a file.
```

Variants that would also fill the gap: _suggests to use a_, _suggests using a_

Author: I think it should 'the use of' and I'll add that.

Reviewer: OK.  Resolving.

If \lstinline!colorSelector = true!, it indicates that an rgb-value selector can be presented for a vector of three elements and generate values 0..255 (the color selector should be useable both for vectors of \lstinline!Integer! and \lstinline!Real!).

> Modifying the \fmtannotationindex{Selector} of \lstinline!loadSelector! or \lstinline!saveSelector! displays a file dialog to select a file.

Reviewer: Yes, this is an improvement.  I'm just not sure about the use of _modifying_ to express what we want in the context of annotations – there are currently no uses of _modifying_ in the chapter.  Perhaps presence of an annotation is a better terminology to use?
```suggestion
Presence of a \fmtannotationindex{Selector} for \lstinline!loadSelector! or \lstinline!saveSelector! displays a file dialog to select a file.
```

Author: I'll reformulate.

Author: I realized that "display" is misleading since the annotation doesn't directly display the dialog.

Reviewer: This looks OK now.  Resolving.

If \lstinline!colorSelector = true!, it indicates that an rgb-value selector can be presented for a vector of three elements and generate values 0..255 (the color selector should be useable both for vectors of \lstinline!Integer! and \lstinline!Real!).

> The presence of \lstinline!loadSelector! or \lstinline!saveSelector! specifying \fmtannotationindex{Selector} provides a file dialog to select a file.

Reviewer: Doesn't _provides_ have the same issue as _displays_?  Borrowing the formulation from `colorSelector`
```suggestion
The presence of a \fmtannotationindex{Selector} for \lstinline!loadSelector! or \lstinline!saveSelector! indicates that a file dialog can be used to select a file.
```

Alternatively, we could reformulate both `colorSelector` and this to use a phrase based on suggesting something instead:
```suggestion
The presence of a \fmtannotationindex{Selector} for \lstinline!loadSelector! or \lstinline!saveSelector! suggests the use of a file dialog to select a file.
```

Along with:
```
Setting \lstinline!colorSelector = true! suggests the use of an rgb-value selector to pick a vector of three elements with values in the range 0..255 (the color selector should be useable both for vectors of \lstinline!Integer! and \lstinline!Real!).
```

Reviewer: This discussion has been continued in response to a later review comment.  Resolving.
